src/System.Device.Gpio/System/Device/Gpio/Drivers/LibgpiodDriver.cs

```diff
@@ -0,0 +1,190 @@
+
+        public LibgpiodDriver() {
+            _pinNumperToPinModes = new Dictionary<int, PinMode>();
+            _pinNumperToProcess = new Dictionary<int, Process>();
+            _waitMilliSeconds = (int)TimeSpan.FromSeconds(1).TotalMilliseconds;
+        }
+        protected internal override void AddCallbackForPinValueChangedEvent(int pinNumber, PinEventTypes eventType, PinChangeEventHandler callback)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected internal override void ClosePin(int pinNumber)
+        {
+            // do nothing_pinNumperToProcess
+        }
+
+        protected internal override int ConvertPinNumberToLogicalNumberingScheme(int pinNumber) => throw new PlatformNotSupportedException("This driver is generic so it can not perform conversions between pin numbering schemes.");
+
+        protected internal override PinMode GetPinMode(int pinNumber)
+        {
+            if (_pinNumperToPinModes.ContainsKey(pinNumber))
+             {
+                return _pinNumperToPinModes[pinNumber];
+            }
+            else
+            {
+                throw new InvalidOperationException("There was an attempt to get a mode to a pin that is not yet open.");
+            }
+        }
+
+        protected internal override bool IsPinModeSupported(int pinNumber, PinMode mode)
+        {
+            // Libgpiod driver does not support pull up or pull down resistors.
+            if (mode == PinMode.InputPullDown || mode == PinMode.InputPullUp)
+                return false;
+            return true;
+        }
+
+        protected internal override void OpenPin(int pinNumber)
+        {
+            // do nothing
+        }
+
+        protected internal override PinValue Read(int pinNumber)
+        {
+            PinValue result = default(PinValue);
+            try
+            {
+                Process myProc = new Process();
+                myProc.StartInfo.FileName = GpioGet;
+                myProc.StartInfo.Arguments = $" {GpioChip0} {pinNumber}";
+                myProc.StartInfo.RedirectStandardOutput = true;
+                myProc.Start();
+                if (HandlProcessExit(myProc, $"{GpioGet} {GpioChip0} {pinNumber}", _waitMilliSeconds))
+                {
+                    string valueContents = myProc.StandardOutput.ReadToEnd();
+                    result = ConvertSysFsValueToPinValue(valueContents);
+                }
+            }
+            catch (UnauthorizedAccessException e)
+            {
+                throw new UnauthorizedAccessException("Reading a pin value requires root permissions.", e);
+            }
+            return result;
+        }
+        private PinValue ConvertSysFsValueToPinValue(string value)
+        {
+            PinValue result;
+            value = value.Trim();
```

Comment:
This feels like an unnecessary allocation.  Can we change this private method to take a char and avoid the string allocation?  If this is copied from the UnixSysFsDriver we should fix it there too

Reply:
I agree - optionally perhaps something like: `return value.IndexOf('1') != -1 ? PinValue.High : PinValue.Low` or just get a char if possible (you should be able to read it directly from the stream and there should be a way of doing that so it ignores whitespaces)